app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+
+  def self.update_user_posts_count(user)
```

Comment:
### Optional
- How about using the `after_save` rails helper method to make the model take the action of updating the posts counter each time the model is being saved to the database!

Reply:
Thanks but I won't add this for now, but it is a good thing to know